core/trino-main/src/main/java/io/trino/operator/output/SkewedPartitionRebalancer.java

```diff
@@ -461,4 +514,42 @@ public boolean equals(Object o)
             return that.id == id;
         }
     }
+
+    // This takes less space than TDigestHistogram since it doesn't include the digest and it's enough for analysis.
+    @JsonInclude(NON_NULL)
+    public record Distribution(long total, double min, double max, double p01, double p05, double p10, double p25, double p50, double p75, double p90, double p95, double p99)
+    {
+        public static Distribution create(TDigestHistogram histogram)
+        {
+            return new Distribution(
+                    histogram.getTotal(),
+                    histogram.getMin(),
+                    histogram.getMax(),
+                    histogram.getP01(),
+                    histogram.getP05(),
+                    histogram.getP10(),
+                    histogram.getP25(),
+                    histogram.getP50(),
+                    histogram.getP75(),
+                    histogram.getP90(),
+                    histogram.getP95(),
+                    histogram.getP99());
+        }
+    }
+
+    /**
+     * Stats for scale writers
+     *
+     * @param partitionsAssigned distribution of partitions assigned to writers or tasks
+     * @param scalingCount number of times scaling happened
```

Comment:
Would be good to also have some kind of data distribution metric between sinks, e.g: to be able to verify that scaling indeed balanced writes.